sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/Schema.java

```diff
@@ -1121,7 +1121,8 @@ public boolean typesEqual(Field other) {
 
     private boolean equivalent(Field otherField, EquivalenceNullablePolicy nullablePolicy) {
       return getName().equals(otherField.getName())
-          && getType().equivalent(otherField.getType(), nullablePolicy);
+          && getType().equivalent(otherField.getType(), nullablePolicy)
+          && getDescription().equals(otherField.getDescription());
```

Comment:
I'm not sure that description should be included here as it's not really part of the type.

Reply:
this is checking whole Field equivalence, not just type. For Type equivalence we have `FieldType.equivalent(..)`,  and `Field.typesEqual(...)` - thoughts?

Reply:
Yes, it is meant to be used to check schema equivalence (where name + type is what matters). I don't think description should be semantic here.

Reply:
removed this and changed for an equality check.